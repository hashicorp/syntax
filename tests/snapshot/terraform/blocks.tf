resource "aws_instance" "web" {
  ami           = "ami-a1b2c3d4"
  instance_type = "t2.micro"
  timeouts {
    create = "60m"
    delete = "2h"
  }
}

// packer type block
// As per https://github.com/hashicorp/hcl/blob/main/hclsyntax/spec.md#identifiers
// a `-` is allowed in the identifier.
// duplicated here to ensure compatibility with generic HCL
post-processor "foo" {}

// dash is not expected at the beginning
-test "foo" {}

// all Unicode word characters are accepted
žblock "foo" {}
bžlock "foo" {}
