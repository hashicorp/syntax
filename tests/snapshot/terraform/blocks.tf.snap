>resource "aws_instance" "web" {
#^^^^^^^^ source.hcl.terraform meta.block.hcl entity.name.type.terraform
#        ^ source.hcl.terraform meta.block.hcl
#         ^^^^^^^^^^^^^^ source.hcl.terraform meta.block.hcl variable.other.enummember.hcl
#                       ^ source.hcl.terraform meta.block.hcl
#                        ^^^^^ source.hcl.terraform meta.block.hcl variable.other.enummember.hcl
#                             ^ source.hcl.terraform meta.block.hcl
#                              ^ source.hcl.terraform meta.block.hcl punctuation.section.block.begin.hcl
>  ami           = "ami-a1b2c3d4"
#^^ source.hcl.terraform meta.block.hcl
#  ^^^ source.hcl.terraform meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#     ^^^^^^^^^^^ source.hcl.terraform meta.block.hcl variable.declaration.hcl
#                ^ source.hcl.terraform meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#                 ^ source.hcl.terraform meta.block.hcl variable.declaration.hcl
#                  ^ source.hcl.terraform meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#                   ^^^^^^^^^^^^ source.hcl.terraform meta.block.hcl string.quoted.double.hcl
#                               ^ source.hcl.terraform meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>  instance_type = "t2.micro"
#^^ source.hcl.terraform meta.block.hcl
#  ^^^^^^^^^^^^^ source.hcl.terraform meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#               ^ source.hcl.terraform meta.block.hcl variable.declaration.hcl
#                ^ source.hcl.terraform meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#                 ^ source.hcl.terraform meta.block.hcl variable.declaration.hcl
#                  ^ source.hcl.terraform meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#                   ^^^^^^^^ source.hcl.terraform meta.block.hcl string.quoted.double.hcl
#                           ^ source.hcl.terraform meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>  timeouts {
#^^ source.hcl.terraform meta.block.hcl
#  ^^^^^^^^ source.hcl.terraform meta.block.hcl meta.block.hcl entity.name.type.hcl
#          ^ source.hcl.terraform meta.block.hcl meta.block.hcl
#           ^ source.hcl.terraform meta.block.hcl meta.block.hcl punctuation.section.block.begin.hcl
>    create = "60m"
#^^^^ source.hcl.terraform meta.block.hcl meta.block.hcl
#    ^^^^^^ source.hcl.terraform meta.block.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#          ^ source.hcl.terraform meta.block.hcl meta.block.hcl variable.declaration.hcl
#           ^ source.hcl.terraform meta.block.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#            ^ source.hcl.terraform meta.block.hcl meta.block.hcl variable.declaration.hcl
#             ^ source.hcl.terraform meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#              ^^^ source.hcl.terraform meta.block.hcl meta.block.hcl string.quoted.double.hcl
#                 ^ source.hcl.terraform meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>    delete = "2h"
#^^^^ source.hcl.terraform meta.block.hcl meta.block.hcl
#    ^^^^^^ source.hcl.terraform meta.block.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#          ^ source.hcl.terraform meta.block.hcl meta.block.hcl variable.declaration.hcl
#           ^ source.hcl.terraform meta.block.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#            ^ source.hcl.terraform meta.block.hcl meta.block.hcl variable.declaration.hcl
#             ^ source.hcl.terraform meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#              ^^ source.hcl.terraform meta.block.hcl meta.block.hcl string.quoted.double.hcl
#                ^ source.hcl.terraform meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>  }
#^^ source.hcl.terraform meta.block.hcl meta.block.hcl
#  ^ source.hcl.terraform meta.block.hcl meta.block.hcl punctuation.section.block.end.hcl
>}
#^ source.hcl.terraform meta.block.hcl punctuation.section.block.end.hcl
>